oz_tree_build/newick/extract_trees.py

```diff
@@ -24,86 +21,85 @@
+    # Fix up situation that would end up generating "(,"
+    if tree_string and tree_string[-1] == "(" and newick_tree[start] == ",":
+        start += 1
+    return newick_tree[start:end]
+
+
 def extract_trees(
     newick_tree,
     target_taxa: Set[str],
-    excluded_taxa: Set[str] = {},
+    excluded_taxa=None,
     included_ancestor_count=0,
 ):
     # We build the subtrees and exclusion lists as we find them and process them
+    if excluded_taxa is None:
+        excluded_taxa = set()
     subtrees = []
     excluded_ranges = []
 
     # Clone the taxa set so we don't modify the original
     target_taxa = set(target_taxa)
 
-    # Keep track of how many ancestors we still need to find among all the subtrees
-    # This is an optimization to avoid processing the whole tree once we've found everything
+    # Keep track of how many ancestors we still need to find among all the subtrees. This
+    # is an optimization to avoid processing the whole tree once we've found everything
     overall_ancestor_needed = 0
 
     for node in parse_tree(newick_tree):
         taxon = node["taxon"]
         ott = node["ott"]
-        node_start_index = node["start"]
-        node_end_index = node["end"]
+        node_start_idx = node["start"]
+        node_end_idx = node["end"]
 
-        # Of any subtree that needs ancestors, add the current taxon to it if it's an ancestor
+        # Of any subtree that needs ancestors, add the current taxon if it's an ancestor
         if overall_ancestor_needed:
             for subtree in subtrees:
-                if subtree["ancestors_needed"] and node_start_index < subtree["start"]:
+                if subtree["ancestors_needed"] and node_start_idx < subtree["start"]:
                     subtree["tree_string"] = "(" + subtree["tree_string"] + ")" + taxon
                     subtree["ancestors_needed"] -= 1
                     overall_ancestor_needed -= 1
 
         # If this taxon or ott is in the excluded list, add it to the excluded ranges
         if taxon in excluded_taxa or ott in excluded_taxa:
             # Use different logic depending on comma position
-            if newick_tree[node_start_index - 1] == ",":
-                # Exclude the comma before the excluded taxon. e.g. (A,B,REMOVE_ME) --> (A,B)
-                excluded_range = (node_start_index - 1, node_end_index)
-            elif newick_tree[node_end_index] == ",":
-                # Exclude the comma after the excluded taxon. e.g. (REMOVE_ME,B,C) --> (B,C)
-                excluded_range = (node_start_index, node_end_index + 1)
+            if newick_tree[node_start_idx - 1] == ",":
+                # Exclude comma before the excluded taxon. e.g. (A,B,REMOVE_ME) --> (A,B)
+                excluded_range = (node_start_idx - 1, node_end_idx)
+            elif newick_tree[node_end_idx] == ",":
+                # Exclude  comma after the excluded taxon. e.g. (REMOVE_ME,B,C) --> (B,C)
```

Comment:
A little strange how Ruff removed the `the` here, leaving 2 spaces. And aside from the spaces, it feels more semantically/grammatically correct with `the` in there.

Reply:
I don't care strongly at all either way. It was all more in the context of reducing the diff to review. But that's ok, I'll just go through it as is. It's not so bad :)

Reply:
Yes, I made quite a lot of changes (>100, I think) in order to get all the ruff checks to pass. In retrospect, I suppose I should have put all the automatic changes into one commit, and all the hand-edited ones in another.